daemon/redis.c

```diff
@@ -1581,8 +2055,48 @@ int redis_restore(struct callmaster *m, struct redis *r) {
 	return ret;
 }
 
+#define JSON_SET_NSTRING(a,b,c,d) do { \
+		sprintf(tmp,a,b); \
+		json_builder_set_member_name (builder, tmp); \
+		ZERO(tmp); \
+		snprintf(tmp,sizeof(d), c,d); \
```

Comment:
I believe this should be `sizeof(tmp)` instead of `sizeof(d)`. The above call to `sprintf` should also be protected with `snprintf` in the same way. The calls to `ZERO` can be omitted. There's no reason not to use `snprintf` everywhere here (including the few more instances further down), so `JSON_SET_STRING` should be removed completely. Sorry for being picky about this.